docs/architecture/adr-050-sign-mode-textual.md

```diff
@@ -0,0 +1,492 @@
+
+### 5. Strings starting with `*` are only shown in Expert mode
+
+Ledger devices have the an Expert mode for advanced users, which needs to be manually activated. Strings starting with the `*` character will only be shown in Expert mode.
+
+### 6. Strings MAY contain `>` characters to denote nesting
+
+Protobuf objects can be arbitrarily complex, containing nested arrays and messages. In order to help the Ledger-signing users, we propose to use the `>` symbol in the beginning of strings to represent nested objects, where each additional `>` represents a new level of nesting.
+
+### 7. Encoding of the Transaction Envelope
+
+We define "transaction envelope" as all data in a transaction that is not in the `TxBody`. Transaction envelope includes fee, signer infos and memo, but don't include `Msg`s. `//` denotes comments and are not shown on the Ledger device.
+
+```
+Chain ID: <string>
+Account number: <uint64>
+*Public Key: <base64_string>
+Sequence: <uint64>
+<TxBody>                                                    // See 8.
+Fee: <coins>                                                // See value renderers for coins encoding.
+*Fee payer: <string>                                        // Skipped if no fee_payer set
+*Fee granter: <string>                                      // Skipped if no fee_granter set
+Memo: <string>                                              // Skipped if no memo set
+*Gas Limit: <uint64>
+*Timeout Height:  <uint64>                                  // Skipped if no timeout_height set
+Tipper: <string>                                            // If there's a tip
+Tip: <string>
+*This transaction has <int> body extension:                 // Skipped if no body extension options
+*<repeated Any>
+*This transaction has <int> body non-critical extensions:   // Skipped if no body non-critical extension options
+*<repeated Any>                                             // See value renderers for Any and array encoding.
+*This transaction has <int> body auth info extensions:      // Skipped if no auth info extension options
+*<repeated Any>
+*This transaction has <int> other signers:                  // Skipped if there is only one signer
+*Signer (<int>/<int>):
+*Public Key: <base64_string>
+*Sequence: <uint64>
+*End of other signers
+```
+
+### 8. Encoding of the Transaction Body
+
+We call "transaction body" the `Tx.TxBody.Messages` field, which is an array of `Any`s. Since messages are widely used, they have a slightly different encoding than usual array of `Any`s (protobuf: `repeated repeated google.protobuf.Any`) described in Annex 1.
+
+```
+This transaction has <int> message:   // Optional 's' for "message" if there's is >1 sdk.Msgs.
+// For each Msg, print the following 2 lines:
+Msg (<int>/<int>): <string>           // E.g. Msg (1/2): bank v1beta1 send coins
```

Comment:
Why not print the fully qualified proto path name?

Reply:
I think we should. It's long, but it will avoid conflicts. Essentially an App can use SDK bank module and it's own custom bank module.

Reply:
> Why not print the fully qualified proto path name?

To make it more user-friendly. It's an optional proto field, if it's not set, then the proto FQ name will be used. See the `type_url` one in the annex.

> I think we should. It's long, but it will avoid conflicts. Essentially an App can use SDK bank module and it's own custom bank module.

These strings are unique per chain, so if a chain has another bank module, it would be e.g. `my_chain bank v1beta1 send coins`